lib/main.ts

```diff
@@ -35,20 +42,16 @@ class AchYouDataProviderEngine
   }
 
   /**
-   * @param selector string userId of user who just sneezed or number sneezeId of sneeze that
-   *  needs editing
+   * @param selector string userId of user who just sneezed a new sneeze or 
+   *  number sneezeId of sneeze that needs editing
    * @param data date and any comments associated with the sneeze
    */
   // Note: this method gets layered over so that you can run `this.set` in the data provider engine,
   // and it will notify update afterward.
-  async setAchYou(selector: string | number, data: Sneeze | User) {
-    if (selector === 'NEWUSER') {
-      logger.log('About to push new user');
-      logger.log(`data.name: ${(data as User).name}`);
-      this.users.push(data as User);
-    }
+  async setSneeze(selector: string | number | Date, data: Sneeze) {
+    // Note: a sneeze can be gotten by date but not set by date so we don't have a case for that
     // Setting a sneeze by userId means that user just sneezed a new sneeze so add it to the data
-    else if (typeof selector === 'string') this.sneezes.push(data as Sneeze);
+    if (typeof selector === 'string') this.sneezes.push(data as Sneeze);
```

Comment:
You don't have to type assert this to `Sneeze` with `as Sneeze` anymore since the data is of type `Sneeze`, right?

Also, if you'd like, you can rename the `data` parameter to `sneeze` or whatever you'd like :) parameter names are not enforced by interfaces. That being said, you are welcome to leave it as well. Either way is great.

Same thing for the `setUser` function below.

Reply:
Good call!